src/plugins/discover/public/application/main/components/top_nav/app_menu_actions/get_alerts.tsx

```diff
@@ -0,0 +1,173 @@
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import type { DataView } from '@kbn/data-plugin/common';
+import { i18n } from '@kbn/i18n';
+import {
+  AppMenuActionId,
+  AppMenuActionSubmenuSecondary,
+  AppMenuActionType,
+} from '@kbn/discover-utils';
+import {
+  AlertConsumers,
+  ES_QUERY_ID,
+  RuleCreationValidConsumer,
+  STACK_ALERTS_FEATURE_ID,
+} from '@kbn/rule-data-utils';
+import { RuleTypeMetaData } from '@kbn/alerting-plugin/common';
+import { DiscoverStateContainer } from '../../../state_management/discover_state';
+import { AppMenuDiscoverParams } from './types';
+
+const EsQueryValidConsumer: RuleCreationValidConsumer[] = [
+  AlertConsumers.INFRASTRUCTURE,
+  AlertConsumers.LOGS,
+  AlertConsumers.OBSERVABILITY,
+  STACK_ALERTS_FEATURE_ID,
+];
+
+interface EsQueryAlertMetaData extends RuleTypeMetaData {
+  isManagementPage?: boolean;
+  adHocDataViewList: DataView[];
+}
+
+const CreateAlertFlyout: React.FC<{
+  getDiscoverParams: () => AppMenuDiscoverParams;
+  onFinishAction: () => void;
+  stateContainer: DiscoverStateContainer;
+}> = React.memo(({ stateContainer, getDiscoverParams, onFinishAction }) => {
+  const query = stateContainer.appState.getState().query;
+
+  const { dataView, services, isEsqlMode, adHocDataViews, onUpdateAdHocDataViews } =
+    getDiscoverParams();
+  const { triggersActionsUi } = services;
+  const timeField = getTimeField(dataView);
+
+  /**
+   * Provides the default parameters used to initialize the new rule
+   */
+  const getParams = useCallback(() => {
+    if (isEsqlMode) {
+      return {
+        searchType: 'esqlQuery',
+        esqlQuery: query,
+        timeField,
+      };
+    }
+    const savedQueryId = stateContainer.appState.getState().savedQuery;
+    return {
+      searchType: 'searchSource',
+      searchConfiguration: stateContainer.savedSearchState
+        .getState()
+        .searchSource.getSerializedFields(),
+      savedQueryId,
+    };
+  }, [isEsqlMode, stateContainer.appState, stateContainer.savedSearchState, query, timeField]);
+
+  const discoverMetadata: EsQueryAlertMetaData = useMemo(
+    () => ({
+      isManagementPage: false,
+      adHocDataViewList: adHocDataViews,
+    }),
+    [adHocDataViews]
+  );
+
+  return triggersActionsUi?.getAddRuleFlyout({
+    metadata: discoverMetadata,
+    consumer: 'alerts',
+    onClose: (_, metadata) => {
+      onUpdateAdHocDataViews(metadata!.adHocDataViewList);
+      onFinishAction();
+    },
+    onSave: async (metadata) => {
+      onUpdateAdHocDataViews(metadata!.adHocDataViewList);
```

Comment:
Is it necessary to call `onFinishAction` here too?

Reply:
We had the same before. Looks like `onClose` would be called from the alerts flyout when `onSave` is called.